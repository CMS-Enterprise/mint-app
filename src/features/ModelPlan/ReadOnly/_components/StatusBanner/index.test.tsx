import React from 'react';
import { createMemoryRouter, RouterProvider } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import { ModelStatus } from 'gql/generated/graphql';

import ReadViewStatusBanner from './index';

describe('ReadViewStatusBanner', () => {
  it('renders correctly and matches snapshot', () => {
    const router = createMemoryRouter(
      [
        {
          path: '/',
          element: (
            <ReadViewStatusBanner
              modelID="123"
              status={ModelStatus.PLAN_DRAFT}
              hasEditAccess
              mostRecentEdit="2024-05-01T12:00:00Z"
              changeHistoryLink
              className="test-class"
            />
          )
        }
      ],
      {
        initialEntries: ['/']
      }
    );

    const { asFragment } = render(<RouterProvider router={router} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders without edit access and without change history link', () => {
    const router = createMemoryRouter(
      [
        {
          path: '/',
          element: (
            <ReadViewStatusBanner
              modelID="123"
              status={ModelStatus.PLAN_DRAFT}
              hasEditAccess={false}
              mostRecentEdit="2024-05-01T12:00:00Z"
              changeHistoryLink={false}
            />
          )
        }
      ],
      {
        initialEntries: ['/']
      }
    );

    render(<RouterProvider router={router} />);

    expect(
      screen.queryByRole('link', { name: /Edit model information/i })
    ).not.toBeInTheDocument();
  });
});
