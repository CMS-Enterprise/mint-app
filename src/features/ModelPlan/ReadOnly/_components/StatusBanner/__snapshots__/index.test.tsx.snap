// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ReadViewStatusBanner > renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="test-class"
    data-testid="task-list-status"
  >
    <div
      class="grid-row desktop:grid-col-12 margin-bottom-2 flex-align-center flex-justify-start"
      data-testid="grid"
    >
      <p
        class="margin-y-0 text-bold margin-right-1"
      >
        Status:
      </p>
      <span
        class="usa-tag mint-tag bg-base text-white margin-right-1 padding-x-1 margin-right-2"
        data-testid="tag"
      >
        Draft Model Plan
      </span>
      <div
        class="grid-row"
        data-testid="grid"
      >
        <div
          class="status-banner__border border-base-light margin-05 margin-right-2 margin-left-1"
        />
        <div
          class="display-flex flex-align-center flex-wrap"
        >
          <a
            class="usa-link display-flex flex-align-center padding-right-2"
            href="/models/123/change-history"
          >
            <svg
              class="usa-icon margin-right-1"
              focusable="false"
              height="1em"
              role="img"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13 3a9 9 0 0 0-9 9H1l3.89 3.89.07.14L9 12H6c0-3.87 3.13-7 7-7s7 3.13 7 7-3.13 7-7 7c-1.93 0-3.68-.79-4.94-2.06l-1.42 1.42A8.954 8.954 0 0 0 13 21a9 9 0 0 0 0-18zm-1 5v5l4.28 2.54.72-1.21-3.5-2.08V8H12z"
              />
            </svg>
            View change history
          </a>
          <div
            class="display-flex flex-align-center flex-wrap margin-right-1"
          >
            <p
              class="margin-y-0 text-normal margin-right-1"
            >
              Model last updated 05/01/2024
            </p>
          </div>
        </div>
        <div
          class="status-banner__border border-base-light margin-05 margin-right-2"
        />
        <div
          class="display-flex flex-align-center"
        >
          <a
            class="usa-link display-flex flex-align-center"
            href="/models/123/collaboration-area"
          >
            <svg
              class="usa-icon margin-right-1"
              focusable="false"
              height="1em"
              role="img"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04a.996.996 0 0 0 0-1.41l-2.34-2.34a.996.996 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
            </svg>
            Edit Model Plan
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
